pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -842,37 +842,42 @@ cdef int64_t parse_iso_format_string(str ts) except? -1:
         object sign_part, integer_part, fractional_part
         list groups = [], units = []
 
+    if ts[0] == "-":
+        sign = -1
+
     iso_regex = re.compile(
-        r"^(-?)P(?!$)(?:"
-        r"(?:(-?)(\d+)(?:[.,](\d{1,9})(?=Y$))?Y)?"
-        r"(?:(-?)(\d+)(?:[.,](\d{1,9})(?=M$))?M)?"
-        r"(?:(-?)(\d+)(?:[.,](\d{1,9})(?=W$))?W)?"
-        r"(?:(-?)(\d+)(?:[.,](\d{1,9})(?=D$))?D)?"
+        # Allow starting with 'P' or '-P', disallow ending on 'P'.
+        r"^-?P(?!$)(?:"
+        # Capture the sign, integer part, and fractional part (but only if it
+        # is the final component).
+        r"(?:(-?)(\d+)(?:[.,](\d{1,9})(?=Y$))?Y)?"  # Years
```

Comment:
Should definitely tighten up these repetitions - `\d{1,9}` is not really necessary